File: apis/python_interface_helpers/stk_tcp_server.py
from stk_data_flow import *
class stk_tcp_server:
	def __init__(self,env,name,id,options,ref=None):
		if ref == None:
			self._df = stk_tcp_server_create_data_flow(env.ref(),name,id,options.ref())
			if self._df == None:
				raise Exception("Failed to create tcp server data flow")
		else:
			self._df = ref
		self._id = stk_get_data_flow_id(self._df)
		self._dfptr = stk_df_ptr_to_ulong_df(self._df)
		stk_data_flow.map(self._dfptr,self);
	def close(self):
		if self._id != None:
			stk_data_flow.unmap(self._dfptr)
			stk_destroy_data_flow(self._df)
	def ref(self):
		return self._df
	def id(self):
		return stk_get_data_flow_id(self._df)
	def send(self,seq,flags):
		return stk_data_flow_send(self._df,seq.ref(),flags)

class stk_tcp_publisher(stk_tcp_server):
	pass

File: apis/ruby_api/stk_tcp_server.rb
require 'stk_data_flow'
require 'stkdata_flow'
require 'stktcp_server'
class Stk_tcp_server
	def initialize(env,name,id,options,ref=nil)
		@_env = env
		if ref.nil?
			@_df = Stktcp_server.stk_tcp_server_create_data_flow(env.ref(),name,id,options.ref())
			raise "Failed to create tcp server data flow" if @_df.nil?
		else
			@_df = ref
		end
		@_id = Stkdata_flow.stk_get_data_flow_id(@_df)
		@_dfptr = Stkdata_flow.stk_df_ptr_to_ulong_df(@_df)
		Stk_data_flow.map(@_dfptr,self);
	end
	def close()
		if !@_id.nil?
			Stk_data_flow.unmap(@_dfptr)
			Stkdata_flow.stk_destroy_data_flow(@_df)
		end
	end
	def ref()
		@_df
	end
	def id()
		Stkdata_flow.stk_get_data_flow_id(@_df)
	end
	def send(seq,flags)
		Stkdata_flow.stk_data_flow_send(@_df,seq.ref(),flags)
	end
	def env()
		@_env
	end
end

class Stk_tcp_publisher < Stk_tcp_server
end
